src/python/yapykaldi/asr/pipeline.py

```diff
@@ -0,0 +1,173 @@
+"""Pipeline manager class"""
+from __future__ import print_function, division, absolute_import, unicode_literals
+from builtins import *
+from threading import Event
+from ..logger import logger
+
+
+class AsrPipeline(object):
```

Comment:
Make this an `AsrPipelineElementBase` subclass as well for extra software architecture hard-on :-). 

It matches the API already and allows for nesting, which is always cool somehow :sweat_smile:

Reply:
Haha yeah! I had to think a lot for making it like this. Will subclass it.

Reply:
I'm going to make this into an issue as there are some attributes that need cleanup by converting it into a property